orderer/common/follower/follower_chain.go

```diff
@@ -500,14 +521,18 @@ func (c *Chain) pullUntilTarget(targetHeight uint64, inspectConfig bool) (uint64
 				return n, errors.Wrapf(err, "failed to append block %d to the ledger", nextBlock.Header.Number)
 			}
 
-			if inspectConfig && protoutil.IsConfigBlock(nextBlock) {
+			if protoutil.IsConfigBlock(nextBlock) {
 				c.logger.Debugf("Pulled blocks from %d to %d, last block is config", firstBlockToPull, nextBlock.Header.Number)
 				c.lastConfig = nextBlock
-				// Re-create the block puller to apply the new config, which may update the endpoints.
-				c.blockPuller.Close()
-				var err error
-				if c.blockPuller, err = c.createBlockPullerFunc(); err != nil { // TODO replace with a factory that has a config block as input
-					return n, errors.Wrapf(err, "failed to re-create block puller from last config,  block number: %d", nextBlock.Header.Number)
+				if errVer := c.blockPullerFactory.UpdateVerifierFromConfigBlock(nextBlock); errVer != nil {
+					return n, errors.Wrapf(errVer, "failed to update verifier from last config,  block number: %d", nextBlock.Header.Number)
+				}
+				if updateEndpoints {
+					endpoints, errEP := cluster.EndpointconfigFromConfigBlock(nextBlock, c.cryptoProvider)
```

Comment:
Style nits here also: why errVer and errEP?

Reply:
done